### PR TITLE
fix(pull): adjust flow after a dataset is pulled

### LINKED ISSUE #635: bug(pull): after pulling a dataset, you should be taken to the most recent version of the dataset
"When pulling another user’s dataset, user lands on workbench with no username/name in the header, and “You are viewing another user’s dataset” message. Goes away after reload or navigate to collection and back (It looks like the fix for this is to select the HEAD commit immediately after pulling)"

### PR DESCRIPTION
**Pull request by ramfox**
Also removes dead code and adjusts e2e tests

closes #635

### CODE REVIEW

File: app/components/collection/Collection.tsx [hunk 1]
```diff
@@ -4,34 +4,27 @@ import { ipcRenderer, shell } from 'electron'
...
 
 import {
   selectFsiPath,
-  selectDetails,
   selectWorkingDatasetIsPublished,
   selectSessionUsername,
   selectShowDetailsBar,
   selectSidebarWidth,
-  selectHasDatasets
+  selectHasDatasets, selectLogLatestVersion
```
Comment (b5) [inline app/components/collection/Collection.tsx:19]: add newline

File: app/selections.ts [hunk 2]
```diff
@@ -116,6 +116,10 @@ function datasetFromDatasetStore (historyDataset: DatasetStore): Dataset {
  *
  */
 
+export function selectLogLatestVersion (state: Store): VersionInfo | undefined {
+  return selectHasHistory(state) ? selectLog(state)[0] : undefined
+}
```
Comment (b5) [inline app/selections.ts:121]: don't  use the ternary operator to return undefined

```suggestion
export function selectLogLatestVersion (state: Store): VersionInfo? {
  if (selectHasHistory(state)) {
    return selectLog(state)[0]
  }
}
```

also, does the optional typescript operator work on function signatures? I'd love to use it as written  above if so

**Review by b5 — APPROVED** [review]
2 nits. otherwise,  delighted to see all this cleanup!

File: app/selections.ts (hunk 2 above)
Comment (ramfox) [reply app/selections.ts:121]: i wonder how many times you are going to tell me not to use the ternary operator w/ undefined before i internalize it 🤦‍♀️

Comment (ramfox) [reply app/selections.ts:121]: also sadly no ? for optional returns, but if you want to read about the discussion surrounding it: https://github.com/microsoft/TypeScript/issues/7426

**Pull request merged**